### ISSUE TITLE
[BUG] The housenumber from custom tig field is not saved to the shipping info and billing address

### ISSUE THREAD

**Issue by glevhen** [tig-nl/postnl-magento2#337]
The housenumber from custom tig field is not saved to the shipping info and billing address, so it's not shown in billing address and shipping address details on the checkout payment step

**To Reproduce**
1. Enable postcode check in the module settings and street lines qty to 3
2. Go to the checkout page and fill shipping info (the selected country should be `NL`), fill postcode and housenumber with any valid address values, for example `7418et` and `7`
3. press `Next` button and go to the payment step 

**Expected result**
The housenumber should be showed in the billing address and shipping address details

**Actual result**
The housenumber should be showed in the billing address and shipping address details

**Workaround**
Describe the temporarily workaround or if none exists write "None"

**Screenshots**
the housenumber is filled on the shipping step
![image](https://user-images.githubusercontent.com/16398798/198588642-9d47f17d-d9ce-4af2-837c-0a03da835dd4.png)
the housenumber is missing on the payment step in billing address details and shipping address info, only street info is showed
![image](https://user-images.githubusercontent.com/16398798/198588751-0d7c416e-d8b5-4fbc-a4ec-b12840efce3e.png)

- Magento version: [e.g. Magento 2.4.5]
- PHP version: [e.g. 8.1]
- TIG PostNL version: [e.g. 1.12.4]

**Comment by tig-jeffreybranderhorst** [comment]
Hey @glevhen ,

I can reproduce this, so I will place this on the backlog.  We are going to fix this, seems like it only a display problem because the order in the backend has a housenumber, but we will fix this. Thank you for submitting this.

Have a great day,
Jeffrey

**Comment by glevhen** [comment]
> Hey @glevhen ,
> 
> I can reproduce this, so I will place this on the backlog. We are going to fix this, seems like it only a display problem because the order in the backend has a housenumber, but we will fix this. Thank you for submitting this.
> 
> Have a great day, Jeffrey

@tig-jeffreybranderhorst you are right about the housenumber in order info, there it is ok, please fix it for checkout payment step, thank you for quick confirming the issue.

Have a nice day

**Comment by Jeffrey-H** [comment]
I can reproduce this issue as well. There seems to be some problems in saving the street lines in the local storage. The checkout displays everything correctly but the house number and addition are not in the Local Storage.

Magento version: [e.g. Magento 2.4.5-p1]
PHP version: [e.g. 7.4]
TIG PostNL version: [e.g. 1.12.4]

**Comment by DavidBuro210** [comment]
This seems to be because of the following commit: 94bb2e8bd677ea94b724f3eec249e88375c556a7

Can't really find a reason as to why these lines were removed, but adding back `<text args="tigHouseNumber()"></text><br/>` into [details.html](https://github.com/tig-nl/postnl-magento2/blob/master/view/frontend/web/template/billing-address/details.html) seems to work.

Haven't thoroughly tested this though, but wanted to share anyway.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @DavidBuro210 ,

Thank you for the update, I will place your comment in the backlog issue so it can be fixed.

Have a great day,
Jeffrey

**Comment by jdereus87** [comment]
We are experiencing this aswell. Could it be given some higher priority since this makes guests in the checkout very unsure about their address information being entered/submitted correctly?

**Comment by tig-jeffreybranderhorst** [comment]
Hi @jdereus87 ,

It will be in our next release, we hope to be able to release it this week.  We are awaiting approval of PostNL for release because they are testing it. 

Have a great day,
Jeffrey

**Comment by Aquive** [comment]
@Jeffrey-H is it possible to show the code for the fix? We could try to solve it temporarily in our project while waiting for the official release

**Comment by Jeffrey-H** [comment]
> @Jeffrey-H is it possible to show the code for the fix? We could try to solve it temporarily in our project while waiting for the official release

@Aquive the only section I could fix is the one above the Terms and Conditions on the checkout page. For mobile users this will probably decrease the confusion but on desktop it's still in the order overview on the right.

[tig_postnl_house_number_337.diff](https://gist.github.com/Jeffrey-H/e564e187e3d48dc028c5f4ee9a85396a)

**Comment by Aquive** [comment]
Thanks @Jeffrey-H I will look into it! I meant to tag @tig-jeffreybranderhorst to be honest, oops!

@tig-jeffreybranderhorst is it possible to show the code for the fix? We could try to solve it temporarily in our project while waiting for the official release.

**Comment by tig-robertgrundeken** [comment]
@Aquive the diff the posted by @Jeffrey-H (thanks!) [tig_postnl_house_number_337.diff](https://gist.github.com/Jeffrey-H/e564e187e3d48dc028c5f4ee9a85396a) is the code that will be changed in v1.12.5 to fix the house number issue. It is actually a revert of an earlier change for when you have multiple postcode suggestion services installed. 

@Aquive Was this what you where looking for to temporary fix the issue?

**Comment by jdereus87** [comment]
I have tested making a patch with the diff file from Robert and running a before & after test. 

Running that file as a patch does indeed restore the front view in the summary of the housenumber.
However for foreign countries (lets say Belgium for ease) it then does not register the street address correctly in the billing address on the order view (backend). The street field as a whole is missing. 

So with the patch we get:
+ Working view of the dutch housenumber on the checkout summary
- We lose the correct registration of the billing address street fields

Without it: 
- We lose the view of the housenumber on the dutch summary view of the checkout
+ We get a correct registration of the billing address on belgian orders.

For now we will do it without the patch and have put a disclaimer below the ship-via container in the checkout summary. 

Tested:
Magento 2.4.4
PostNL: 1.12.4
Native checkout
